sqlobject/src/main/java/org/jdbi/v3/sqlobject/config/internal/UseTemplateEngineImpl.java

```diff
@@ -14,56 +14,34 @@
+    public void configureForType(ConfigRegistry registry, Annotation annotation, Class<?> sqlObjectType) {
         UseTemplateEngine anno = (UseTemplateEngine) annotation;
-        try {
-            final TemplateEngine templateEngine = instantiate(anno.value(), sqlObjectType, method);
-            registry.get(SqlStatements.class).setTemplateEngine(templateEngine);
-        } catch (Exception e) {
-            throw new IllegalStateException(e);
-        }
+        TemplateEngine engine = instantiate(anno.value());
+        registry.get(SqlStatements.class).setTemplateEngine(engine);
     }
 
     @Override
-    public void configureForType(ConfigRegistry registry, Annotation annotation, Class<?> sqlObjectType) {
-        UseTemplateEngine anno = (UseTemplateEngine) annotation;
-        try {
-            final TemplateEngine templateEngine = instantiate(anno.value(), sqlObjectType, null);
-            registry.get(SqlStatements.class).setTemplateEngine(templateEngine);
-        } catch (Exception e) {
-            throw new IllegalStateException(e);
-        }
+    public void configureForMethod(ConfigRegistry registry, Annotation annotation, Class<?> sqlObjectType, Method method) {
+        configureForType(registry, annotation, sqlObjectType);
     }
 
-    private TemplateEngine instantiate(Class<? extends TemplateEngine> value,
-                                       Class<?> sqlObjectType,
-                                       Method m) throws Exception {
+    private static <T extends TemplateEngine> T instantiate(Class<T> clazz) {
         try {
-            Constructor<? extends TemplateEngine> noArg = value.getConstructor();
-            return noArg.newInstance();
-        } catch (NoSuchMethodException e) {
-            try {
-                Constructor<? extends TemplateEngine> classArg = value.getConstructor(Class.class);
-                return classArg.newInstance(sqlObjectType);
-            } catch (NoSuchMethodException e1) {
-                if (m != null) {
-                    Constructor<? extends TemplateEngine> constructor = value.getConstructor(Class.class,
-                                                                                            Method.class);
-                    return constructor.newInstance(sqlObjectType, m);
-                }
-                throw new IllegalStateException("Unable to instantiate, no viable constructor " + value.getName());
-            }
+            return clazz.getConstructor().newInstance();
+        } catch (NoSuchMethodException ignored) {
+            throw new RuntimeException(clazz.getSimpleName() + " does not have a default constructor and thus cannot be used with @" + UseTemplateEngine.class.getSimpleName());
+        } catch (IllegalAccessException | InstantiationException | InvocationTargetException e) {
+            throw new RuntimeException(e);
```

Comment:
When I suggested getting rid of the `(Class)` and `(Class, Method)` variants, I did not realize it was existing, documented behavior. This behavior should be reinstated--my bad.